cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java

```diff
@@ -612,11 +614,13 @@ private String getWrapTemplateWithSchemaDetection(String whereField,Table target
    * @param targetTable  where to search
    * @return the ForeignKeys if found, null otherwise
    */
-  private ForeignKeys findForeignKey(String field, Table targetTable) {
-    String fieldName = dbTable.getTableName() + field;
-    for (ForeignKeys key : targetTable.getForeignKeys()) {
-      if (fieldName.equalsIgnoreCase(key.getFieldName())) {
-        return key;
+  private ForeignKeys findForeignKey(String field, Table targetTable, Table currentTable) {
```

Comment:
Can you change to natural order of the parameters: Table currentTable, String field, Table targetTable?